### PR TITLE
Define starting rules for base configuration

### PR DESCRIPTION
**Pull request by BrianSipple**
@Ticketfly/frontenders 

This PR defines a small initial ruleset for what would be our base ESLint configuration. 

We can continue to make changes here as we go along, but I wanted to start with a setup that included a few options — and provided a clean, understandable structure for making changes in the future.

With that in mind, I defined extensions to `eslint:recommended` — broken out by [section](http://eslint.org/docs/rules/#possible-errors) and (choosing from the ESLint's available ruleset)[http://eslint.org/docs/rules/]

(Also note the alphabetical ordering, with the official description of the rule stated as a comment 😄).

A good goal might also be to have explicitly defined rules for _all_ ESLint options at some point, but for the time being, I only focused on things that  `eslint:recommended` wasn't already giving us.

As noted in the README, the goal here is to have a "Base" configuration that other Ticketfly projects can import from, picking and choosing the categories they'd like, if they don't want everything, and then applying their own, environment-specific settings at a project level (allowing special globals, for example).

### CODE REVIEW

File: rules/best-practices.js [hunk 1]
```diff
@@ -5,4 +5,42 @@
...
+
+    // disallow use of new operator when not part of the assignment or comparison
+    'no-new': 'error',
+
+    // require immediate function invocation to be wrapped in parentheses
+    // http://eslint.org/docs/rules/wrap-iife.html
+    'wrap-iife': ['error', 'outside'],  // use this style: ``` var x = (function () { return { y: 1 };}()); ```,
+
+    // require or disallow Yoda conditions
+    yoda: 'error'
```
Comment (melissaroman) [inline rules/best-practices.js:44]: I didn't know "Yoda" conditions were a thing. 😂  (Already learned something this morning!)

File: rules/stylistic-issues.js [hunk 2]
```diff
@@ -5,4 +5,64 @@
...
+      }
+    }],
+
+    // disallow mixed 'LF' and 'CRLF' as linebreaks
+    // http://eslint.org/docs/rules/linebreak-style
+    'linebreak-style': ['error', 'unix'],
+
+    // disallow use of unary operators, ++ and --
+    // http://eslint.org/docs/rules/no-plusplus
+    'no-plusplus': 'error',
```
Comment (zhujy8833) [inline rules/stylistic-issues.js]: @BrianSipple Does this disallow `for(i = 0; i < length; i++)`? Should we use option `"allowForLoopAfterthoughts": true` ? - http://eslint.org/docs/rules/no-plusplus

Comment (BrianSipple) [reply rules/stylistic-issues.js]: Nice catch! 100% Agree.

**Comment by zhujy8833** [comment]
@BrianSipple LGTM! I don't have access to a "merge" button, could you check if there is a permission config needs to set up?

**Pull request merged**